### PR TITLE
Catch dnn_hidden_units to avoid obsture error message

### PR DESCRIPTION
**Pull request by terrytangyuan**
Not providing `dnn_hidden_units` in for dnn linear combined model would cause the following error message that's not useful for users.

```
Type'NoneType' object is not iterable
```

### CODE REVIEW

**Comment by terrytangyuan** [comment]
Test failure due to timeout. Will be fixed in #7180

File: tensorflow/contrib/learn/python/learn/estimators/dnn_linear_combined.py [hunk 1]
```diff
@@ -1058,4 +1061,4 @@ def default_input_fn(unused_estimator, examples):
         signature_fn=signature_fn or export.regression_signature_fn,
         prediction_key=prediction_key.PredictionKey.SCORES,
         default_batch_size=default_batch_size,
-        exports_to_keep=exports_to_keep)
+        exports_to_keep=exports_to_keep)
```
Comment (rmlarsen) [inline tensorflow/contrib/learn/python/learn/estimators/dnn_linear_combined.py]: Please add a newline at the end of the file.

**Review by rmlarsen — CHANGES_REQUESTED** [review]

**Comment by terrytangyuan** [comment]
@rmlarsen Done. Thanks!

File: tensorflow/contrib/learn/python/learn/estimators/dnn_linear_combined.py [hunk 2]
```diff
@@ -191,6 +191,9 @@ def _dnn_linear_combined_model_fn(features, labels, mode, params, config=None):
   if not dnn_feature_columns:
     dnn_logits = None
   else:
+    if not dnn_hidden_units:
+      raise ValueError(
```
Comment (rmlarsen) [inline tensorflow/contrib/learn/python/learn/estimators/dnn_linear_combined.py]: Could you add a unit test for this?

**Review by rmlarsen — CHANGES_REQUESTED** [review]
Could you add a unit test for the new exception?

File: tensorflow/contrib/learn/python/learn/estimators/dnn_linear_combined.py (hunk 2 above)
Comment (terrytangyuan) [inline tensorflow/contrib/learn/python/learn/estimators/dnn_linear_combined.py]: Good call. Added.

**Review by rmlarsen — APPROVED** [review]

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by terrytangyuan** [comment]
Seems like an irrelevant issue that Martin is fixing

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Pull request merged**